drivers/npmx/npmx_trigger.c

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <npmx_driver.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(NPMX_TRIGGER, CONFIG_NPMX_LOG_LEVEL);
+
+/* Callback for active sense pin from NPMX device */
+static void npmx_gpio_callback(const struct device *gpio_dev,
+			       struct gpio_callback *cb,
+			       uint32_t pins)
+{
+	struct npmx_data *data = CONTAINER_OF(cb, struct npmx_data, gpio_cb);
+	const struct device *npmx_dev = data->dev;
+	const struct npmx_config *config = npmx_dev->config;
+
+	gpio_pin_interrupt_configure_dt(&config->int_gpio, GPIO_INT_DISABLE);
+	k_work_submit(&data->work);
+}
+
+static void npmx_work_cb(struct k_work *work)
+{
+	struct npmx_data *data = CONTAINER_OF(work, struct npmx_data, work);
+	const struct device *npmx_dev = data->dev;
+	const struct npmx_config *config = npmx_dev->config;
+
+	npmx_instance_t *npmx_instance = &data->npmx_instance;
+
+	npmx_interrupt(npmx_instance);
+
+	npmx_proc(npmx_instance);
+
+	gpio_pin_interrupt_configure_dt(&config->int_gpio, GPIO_INT_LEVEL_HIGH);
+}
+
+int npmx_gpio_interrupt_init(const struct device *dev)
```

Comment:
Why this function is defined in separate file if it is declared in `npmx_driver.h`?

Reply:
Zephyr coding standard, I did similar to BH1749 sensor